lib/utils/injectRefreshEntry.js

```diff
@@ -29,11 +30,23 @@ function isSocketEntry(entry) {
  * Injects an entry to the bundle for react-refresh.
  * @param {WebpackEntry} [originalEntry] A Webpack entry object.
  * @param {import('../types').NormalizedPluginOptions} options Configuration options for this plugin.
+ * @param {import('webpack').Compiler} compiler A webpack compiler object.
  * @returns {WebpackEntry} An injected entry object.
  */
-function injectRefreshEntry(originalEntry, options) {
+function injectRefreshEntry(originalEntry, options, compiler) {
   /** @type {Record<string, *>} */
   let resourceQuery = {};
+
+  if (compiler && compiler.options && compiler.options.devServer) {
```

Comment:
Not sure if depending on `options.devServer` is the best solution here, dev server could also be called from CLI or by `new WebpackDevServer()` in that case `options.devServer` might be undefined

Reply:
cli args get merged into the `devServer` object, so it appears safe to use this

(to test, I used `port:8080` in my devServer config, but then ran the cli with `--port 9090`. the `devServer` on this line had `port: 9090`

Reply:
also, I think it's best to use whatever `devServer` defaults to, and then still keep the options for `options.overlay` to overwrite it when you want to have that extra flexibility, which I think my code here does (note a couple of lines down where options.overlay overwrites resourceQuery)